inspirehep/modules/records/texkeys.py

```diff
@@ -0,0 +1,229 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2017 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Texkeys util functions."""
+
+from __future__ import absolute_import, division, print_function
+
+from datetime import date
+from itertools import chain
+import random
+import re
+import string
+
+from unidecode import unidecode
+
+from invenio_db import db
+from invenio_pidstore.errors import PIDDoesNotExistError
+from invenio_pidstore.models import PersistentIdentifier, PIDStatus, RecordIdentifier
+from invenio_pidstore.providers.base import BaseProvider
+
+
+from inspirehep.modules.records.api import InspireRecord
+from inspirehep.utils.date import create_earliest_date
+from inspirehep.utils.helpers import force_list
+from inspirehep.utils.record import get_value
+
+
+def generate_texkey(record):
+    new_texkey = ''
+    first_part = first_part_generator(record)
+    if first_part:
+        new_texkey = first_part + ':' + second_part_generator(record) + third_part_generator()
+    return new_texkey
+
+
+def first_part_generator(record):
+    if len(record.get('authors', [])) in range(1, 10):
+        first_part = sanitize(record.get('authors')[0]['full_name'])
+    elif len(record.get('collaborations', [])):
+        first_part = sanitize(record.get('collaborations')[0]['value'])
+    elif len(record.get('corporate_author', [])):
+        first_part = sanitize(record.get('corporate_author')[0])
+    elif 'proceedings' in record.get('document_type', []):
+        first_part = 'proceedings'
+    elif len(record.get('authors', [])):
+        first_part = sanitize(record.get('authors')[0]['full_name'])
+    else:
+        first_part = ''
+
+    return first_part
+
+
+def second_part_generator(record):
+    return get_year_from_record(record)
+
+
+def get_three_random_letters():
+    letters = string.letters.lower()
```

Comment:
You actually want [`strings.ascii_lowercase`](https://github.com/inspirehep/inspire-next/issues/2453) here.